R/getStepwiseDivergence.R

```diff
@@ -222,3 +233,36 @@ setMethod("getTimeDivergence",
     return(x)
 
 }
+
+############################ HELPER FUNCTION(S) ################################
+
+.get_mat_from_se <- function (x, assay_name, altexp) {
+    # non-empty string
+    if(!.is_non_empty_string(assay_name)){
+        stop("'assay_name' must be a non-empty single character value.",
+             call. = FALSE)
+    }
+    if (!is.null(altexp)) {
+        x <- altExp(x, altexp)
+    }
+    mat <- NULL
+    if (assay_name %in% assayNames(x)) {
+        mat <- t(assay(x, assay_name))
+    }
+    # x has to be an instance of SingleCellExperiment to have a reducedDim slot
+    if (is(x, "SingleCellExperiment") && assay_name %in% reducedDimNames(x)) {
+        if (!is.null(mat))
+            # if present in both assay and reducedDim
+            warning("'assay_name' was present in assay and reducedDim. ",
+                    "'assay_name' is picked from assay.",
+                    call. = FALSE)
+        else
+            # if not present in assay but in reducedDim
+            mat <- SingleCellExperiment::reducedDim(x, assay_name)
+    }
+    # not present otherwise
+    if (is.null(mat))
+        stop("'assay_name' is not present.",
+             call. = FALSE)
```

Comment:
Isn't this something that could be checked already on the very first lines? If assay_name is not in assayNames and not in reducedDimNames then one can immediately stop with this warning? 

Doing this at the end via investigation of matrix contents is unnecessarily indirect.

Reply:
I guess that's what's going on atm (or perhaps I didn't understand the comment right).
first check name in assay (and pick it up already if so)
second check if in reducedDim (if mat is picked already, then throw warning otherwise pick it up if positive)
Third if nothing was picked then stop.

Reply:
I mean, could it be checked at the very first lines of this function already, that an assay (or reducedDim) with such name exists in the named altExp? And then proceed to pick the object that had that name.

-> Instead of running the whole function with various if/else/pick statements, and only in the end noticing that the result is still empty.